### ISSUE TITLE
Loss of Wi-Fi connection during HTTP POST is handled incorrectly and results in a reboot

### ISSUE THREAD

**Issue by TheSomeMan** [ruuvi/ruuvi.gateway_esp.c#1002]
```
I (493668) transport_base: [adv_post_task] [network.ruuvi.com] Get TLS saved session for ssl=0x3fff9a64: found at idx=1
I (493724) transport_base: [adv_post_task] [network.ruuvi.com] Reuse saved TLS session ticket for host
I (493727) esp-tls: [adv_post_task] esp_tls_init: tls=0x3fffbcb4
I (493737) ADV_POST_TASK: [adv_post_task/23] Advs cnt: 806
I (493886) esp-tls: [adv_post_task] Hostname 'network.ruuvi.com' resolved to 18.157.65.32
I (493937) esp-tls: esp_tls_tcp_connect: Open socket 55
I (494210) ADV_POST_TASK: [adv_post_task/23] Advs cnt: 43
I (494244) esp-tls-mbedtls: [adv_post_task]: [network.ruuvi.com] Configure size of TLS I/O buffers: in_content_len=8192, out_content_len=4096
I (495150) esp-x509-crt-bundle: Certificate validated
I (495211) esp-tls-mbedtls: [adv_post_task]: [] SSL handshake success, TLS version: TLSv1.3
I (495212) transport_base: [adv_post_task] [] Unlock TLS saved session for ssl=0x3fff9a64
I (495218) transport_base: [adv_post_task] Free TLS saved session for ssl=0x3fff9a64, session=0x3fff982c: network.ruuvi.com
I (495240) ADV_POST_TASK: [adv_post_task/23] Advs cnt: 60
I (500510) ruuvi_gateway: [main/1] free heap: 8164 .. 53100
I (501223) wifi:bcn_timout,ap_probe_send_start
I (503727) wifi:ap_probe_send over, resett wifi status to disassoc
I (503728) wifi:state: run -> init (c800)
I (503728) wifi:pm stop, total sleep time: 0 us / 495551038 us

I (503732) wifi:new:<8,0>, old:<8,0>, ap:<255,255>, sta:<8,0>, prof:3
I (503743) wifi_manager: [sys_evt/20] WIFI_EVENT_STA_DISCONNECTED, reason=200 (BEACON_TIMEOUT)
E (503748) transport_base: [adv_post_task] base_poll_write: [] poll_write select error 113 (Software caused connection abort), fd = 55
W (503760) transport_base: [adv_post_task] ssl_write: [] Poll timeout or error, errno=11 (No more processes), fd=55, timeout_ms=15000
I (503774) ADV_POST_TASK: [adv_post_task/23] Got ADV_POST_SIG_RETRANSMIT2
I (503780) ADV_POST_TASK: [adv_post_task/23] Advs cnt: 835
E (503789) transport_base: [adv_post_task] base_poll_write: [] poll_write select error 0 (ESP_OK), fd = 55
W (503797) transport_base: [adv_post_task] ssl_write: [] Poll timeout or error, errno=11 (No more processes), fd=55, timeout_ms=15000
E (503814) transport_base: [adv_mqtt_task] base_poll_write: [] poll_write select error 113 (Software caused connection abort), fd = 54
W (503822) transport_base: [adv_mqtt_task] ssl_write: [] Poll timeout or error, errno=0 (ESP_OK), fd=54, timeout_ms=9000
E (503824) transport_base: [adv_post_task] base_poll_write: [] poll_write select error 0 (ESP_OK), fd = 55
E (503835) MQTT_CLIENT: Writing failed: errno=0
W (503845) transport_base: [adv_post_task] ssl_write: [] Poll timeout or error, errno=11 (No more processes), fd=55, timeout_ms=15000
E (503850) MQTT_CLIENT: [adv_mqtt_task] esp_mqtt_client_dispatch_transport_error: sock_errno=113 (Software caused connection abort)
E (503875) MQTT: [adv_mqtt_task/23] ./main/mqtt.c:335 {mqtt_get_error_code}: MQTT_EVENT_ERROR (MQTT_ERROR_TYPE_TCP_TRANSPORT): 113 (Software caused connection abort)
I (503890) gw_status: [adv_mqtt_task/23] MQTT connection error: CONNECT
I (503897) transport_base: [adv_mqtt_task] base_close: [] tls=0x3ffe44a0
I (503904) esp-tls: [adv_mqtt_task] esp_tls_conn_destroy: [] tls=0x3ffe44a0, sockfd=54
E (503916) transport_base: [adv_post_task] base_poll_write: [] poll_write select error 0 (ESP_OK), fd = 55
W (503924) transport_base: [adv_post_task] ssl_write: [] Poll timeout or error, errno=11 (No more processes), fd=55, timeout_ms=15000
I (503937) MQTT: [adv_mqtt_task/23] MQTT_EVENT_DISCONNECTED
I (503942) gw_status: [adv_mqtt_task/23] MQTT disconnected
I (503948) LEDS: [adv_mqtt_task/23] Notify: MQTT(1) disconnected
W (503955) MQTT_CLIENT: Publish: Losing qos0 data when client not connected
E (503963) ADV_MQTT_TASK: [adv_mqtt_task/23] ./main/adv_mqtt_signals.c:115 {adv_mqtt_handle_sig_recv_adv}: mqtt_publish_adv failed
I (503985) LEDS: [leds_task/8] LEDS_TASK_SIG_ON_EV_MQTT1_DISCONNECTED (ready=1)
I (503985) wifi_manager: [wifi_manager/5] MESSAGE: EVENT_STA_DISCONNECTED with Reason code: 200 (BEACON_TIMEOUT), event_bits=0x3003
E (503987) transport_base: [adv_post_task] base_poll_write: [] poll_write select error 0 (ESP_OK), fd = 55
W (504006) transport_base: [adv_post_task] ssl_write: [] Poll timeout or error, errno=11 (No more processes), fd=55, timeout_ms=15000
I (504022) LEDS: [leds_task/8] LEDS: New blinking mode: GGGGGGGGG-
I (504026) wifi_manager: [wifi_manager/5] lost connection
I (504032) wifi_manager: [wifi_manager/5] wifi_handle_ev_sta_disconnected: activate reconnection after timeout
I (504043) wifi_manager: [wifi_manager/5] Set STA IP String to: 0.0.0.0
I (504050) wifi_manager: [wifi_manager/5] WIFI_MANAGER:EV_STATE: Clear WIFI_MANAGER_WIFI_CONNECTED_BIT | WIFI_MANAGER_ETH_CONNECTED_BIT
W (504063) network: [wifi_manager/5] Wifi disconnected
I (504068) LEDS: [leds_task/8] LEDS_TASK_SIG_ON_EV_NETWORK_DISCONNECTED (ready=1)
E (504070) transport_base: [adv_post_task] base_poll_write: [] poll_write select error 0 (ESP_OK), fd = 55
W (504087) transport_base: [adv_post_task] ssl_write: [] Poll timeout or error, errno=11 (No more processes), fd=55, timeout_ms=15000
I (504101) ADV_POST_TASK: [adv_post_task/23] Handle event: NETWORK_DISCONNECTED
I (504111) ruuvi_gateway: [main/1] ### Handle event: NETWORK_DISCONNECTED
I (504112) TIME: [time_task/1] ### Deactivate SNTP time synchronization
I (504116) ruuvi_gateway: [main/1] ### Stop mDNS
E (504150) transport_base: [adv_post_task] base_poll_write: [] poll_write select error 0 (ESP_OK), fd = 55
W (504151) transport_base: [adv_post_task] ssl_write: [] Poll timeout or error, errno=11 (No more processes), fd=55, timeout_ms=15000
I (504210) ADV_POST_TASK: [adv_post_task/23] Advs cnt: 46
E (504213) transport_base: [adv_post_task] base_poll_write: [] poll_write select error 0 (ESP_OK), fd = 55
W (504215) transport_base: [adv_post_task] ssl_write: [] Poll timeout or error, errno=11 (No more processes), fd=55, timeout_ms=15000
E (504279) transport_base: [adv_post_task] base_poll_write: [] poll_write select error 0 (ESP_OK), fd = 55
W (504279) transport_base: [adv_post_task] ssl_write: [] Poll timeout or error, errno=11 (No more processes), fd=55, timeout_ms=15000
E (504342) transport_base: [adv_post_task] base_poll_write: [] poll_write select error 0 (ESP_OK), fd = 55
W (504342) transport_base: [adv_post_task] ssl_write: [] Poll timeout or error, errno=11 (No more processes), fd=55, timeout_ms=15000
E (504405) transport_base: [adv_post_task] base_poll_write: [] poll_write select error 0 (ESP_OK), fd = 55
W (504405) transport_base: [adv_post_task] ssl_write: [] Poll timeout or error, errno=11 (No more processes), fd=55, timeout_ms=15000
```
...

```
W (588413) transport_base: [adv_post_task] ssl_write: [] Poll timeout or error, errno=11 (No more processes), fd=55, timeout_ms=15000
E (588475) transport_base: [adv_post_task] base_poll_write: [] poll_write select error 0 (ESP_OK), fd = 55
W (588476) transport_base: [adv_post_task] ssl_write: [] Poll timeout or error, errno=11 (No more processes), fd=55, timeout_ms=15000
I (588510) ruuvi_gateway: [main/1] free heap: 12644 .. 17572
E (588511) ruuvi_gateway: [main/1] ./main/main_loop.c:216 {main_task_handle_sig_log_heap_usage}: Only 12KiB of free memory left - probably due to a memory leak. Reboot the Gateway.
I (588524) reset_task: [main/1] Low memory
I (588527) wifi_manager: [main/1] Send msg: ORDER_STOP_AND_DESTROY
I (588534) wifi_manager: [wifi_manager/5] Got msg: ORDER_STOP_AND_DESTROY
```

**Comment by TheSomeMan** [comment]
It's a duplicate of task #1012.
Fixed in commit [2168cec8f5b6905adaed2ea4d4ed9d3dbce41d45]

**Comment by markoaamunkajo** [comment]
Confirmed in discussion that this issue is not seen
